plugins/inputs/minecraft/minecraft.go

```diff
@@ -0,0 +1,137 @@
+package minecraft
+
+// minecraft.go
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+const sampleConfig = `
+  # server address for minecraft
+  server = "localhost"
+  # port for RCON
+  port = "25575"
+  # password RCON for mincraft server
+  password = "replace_me"
+`
+
+// Client is an interface for a client which gathers data from a minecraft server
+type Client interface {
+	Gather() ([]string, error)
+}
+
+// Minecraft represents a connection to a minecraft server
+type Minecraft struct {
+	Server   string
+	Port     string
+	Password string
+	client   Client
+}
+
+// Description gives a brief description.
+func (s *Minecraft) Description() string {
+	return "it collects stats from Minecraft servers"
+}
+
+// SampleConfig returns our sampleConfig.
+func (s *Minecraft) SampleConfig() string {
+	return sampleConfig
+}
+
+// Gather uses the RCON protocal to collect playerName and
+// scoreboard stats from a minecraft server.
+func (s *Minecraft) Gather(acc telegraf.Accumulator) error {
+	if s.client == nil {
+		var err error
+		s.client, err = NewRCON(s.Server, s.Port, s.Password)
+		if err != nil {
+			return err
+		}
+	}
+
+	scores, err := s.client.Gather()
+	if err != nil {
+		return err
+	}
+
+	for _, score := range scores {
+		playerName, err := ParsePlayerName(score)
+		if err != nil {
+			return err
+		}
+		tags := map[string]string{
+			"playerName": playerName,
+			"server":     s.Server,
+		}
+
+		stats, err := ParseScoreboard(score)
+		if err != nil {
+			return err
+		}
+		var fields = make(map[string]interface{}, len(stats))
+		for _, stat := range stats {
+			fields[stat.Name] = stat.Value
+		}
+
+		acc.AddFields("minecraft", fields, tags)
+	}
+
+	return nil
+}
+
+// ParsePlayerName takes an input string from rcon, to parse
+// the playerName.
+func ParsePlayerName(input string) (string, error) {
+	var re = regexp.MustCompile(`for\s(.*):-`)
```

Comment:
Should these regex be package level so they are only compiled once?

I suggest something along the lines of `for\s([^:]+):-`